1.data-exploration/scripts/nbconverted/test-train_split.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import sys
+import pathlib
+import numpy as np
+import pandas as pd
+sys.path.insert(0, "../0.data-download/scripts/")
+from data_loader import load_data
+from sklearn.model_selection import train_test_split
+
+
+# In[2]:
+
+
+# load all of the data 
+data_directory = "../0.data-download/data/"
+dfs = load_data(data_directory, adult_or_pediatric = "all")
+dependency_df = dfs[1]
+sample_df = dfs[0]
+
+
+# In[3]:
+
+
+# assign age_categories and sex columns to the dependency dataframe as a single column
+presplit_dependency_df = dependency_df.assign(age_and_sex = sample_df.age_categories.astype(str) + "_" + sample_df.sex.astype(str))
+presplit_dependency_df
+
+
+# In[4]:
+
+
+groups = sample_df.groupby("age_categories")
+df_list = []
+for name, df in groups:
+    
+    # only looking for samples that contain Adult or Pediatric information
+    if name == "Adult" or name == "Pediatric":
+        df_list.append(df)
+        
+# merge dataframes through concatentation 
+new_df = pd.concat(df_list, axis=0)
+new_df.reset_index(drop=True)
+
+
+# In[5]:
+
+
+# creating a sample data frame that onl contains the desired information of sex and age category
+ref_df = new_df[["DepMap_ID", "sex", "age_categories"]]
+ref_df.reset_index(drop=True)
+ref_df = ref_df.set_index("DepMap_ID")
+ref_df = ref_df.sort_index(ascending=True)
+ref_df = ref_df.reset_index()
```

Comment:
I'm not sure if you need to do this.

Think about what you actually need for train test split. Why do you need this logic?

Reply:
(sorry, this comment is supposed to refer to lines 53-70)